lib/bamboo/mailer.ex

```diff
@@ -184,8 +184,7 @@ defmodule Bamboo.Mailer do
   end
 
   @doc false
-  def parse_opts(mailer, opts) do
-    otp_app = Keyword.fetch!(opts, :otp_app)
+  def parse_opts(mailer, otp_app) do
```

Comment:
❓ This makes it so that the omitting `:otp_app` in a `Mailer` is caught during compile time, but it changes a public method signature. Can add an additional clause `def parse_opts(mailer, opts, otp_app)` if concerned